src/main/java/cubicchunks/asm/mixin/selectable/common/MixinPathNavigate.java

```diff
@@ -0,0 +1,102 @@
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package cubicchunks.asm.mixin.selectable.common;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import mcp.MethodsReturnNonnullByDefault;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLiving;
+import net.minecraft.pathfinding.NodeProcessor;
+import net.minecraft.pathfinding.Path;
+import net.minecraft.pathfinding.PathNavigate;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.ChunkCache;
+import net.minecraft.world.World;
+
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+@Mixin(PathNavigate.class)
+public abstract class MixinPathNavigate {
+
+    @Shadow protected EntityLiving entity;
+    @Shadow protected NodeProcessor nodeProcessor;
+    @Shadow protected Path currentPath;
+    @Shadow protected World world;
+
+    @Shadow
+    protected abstract Vec3d getEntityPosition();
+
+    @Redirect(method = "getPathToPos", at = @At(value = "NEW", target = "net/minecraft/world/ChunkCache"))
+    private ChunkCache newChunkCacheToPosRedirect(World worldIn, BlockPos posFromIn, BlockPos posToIn, int subIn, BlockPos target) {
+        int x1 = (int) entity.posX;
+        int y1 = (int) entity.posY;
+        int z1 = (int) entity.posZ;
+        int x2 = target.getX();
+        int y2 = target.getY();
+        int z2 = target.getZ();
+        return new ChunkCache(worldIn, new BlockPos(Math.min(x1, x2), Math.min(y1, y2), Math.min(z1, z2)),
+                new BlockPos(Math.max(x1, x2), Math.max(y1, y2), Math.max(z1, z2)), 1);
```

Comment:
Are you sure this won't cause MC to access parts of ChunkCache that aren't there?

It's reasonable to expect it to get blocks from outside of the range between target and current position to get around some obstacles.

Reply:
Yes, it will. But it will not cause any troubles because "air" state returned from cache will be marked as non-walkable for ground moving entities.

Reply:
But it means that there will be obstacles that they can't walk around that they normally would walk around. And in the edge case where the entity is directly aligned on some axis with the player, it will be able to walk ONLY in straight line.

Reply:
Path finder is quite dumb. In a range of 16 blocks entity will stay behind fence without attempts to walk around it.
Here is a video with related mixin disabled:
https://youtu.be/FbY_XXMkAl0

Reply:
The entity won't attempt to walk around it because it has limits to how far it attempts to go. It doesn't try to go around because there is no path around in the area is searches. Your code limits it's attempts to walk around a fence to the chunks that are directly between you and the entity.

Reply:
Yes.
Your conclusion? Is this PR is inappropriate in any form including mixins disabled by default?

Reply:
If it's disabled by default it's ok as long as there is description that says it may change vanilla behavior, making mobs slightly dumber.